release-engineering/selection.md

```diff
@@ -0,0 +1,85 @@
+Release Managers group is a group of Kubernetes contributors (under the Release
+Engineering subproject of SIG Release) which is tasked with publishing
+Kubernetes release artifacts, as well as maintaining the processes and tooling
+to do so.
+
+- [Expectations](#expectations)
+- [Identification](#identification)
+- [Selection](#selection)
+- [Promotion](#promotion)
+
+## Expectations
+
+(From ["Becoming a Release Manager Associate"][becoming-a-release-manager-associate])
+
+Contributors can become Associates by demonstrating the following:
+
+- consistent participation, including 6-12 months of active release
+  engineering-related work
+- experience fulfilling a technical lead role on the Release Team during a
+  release cycle
+  - this experience provides a solid baseline for understanding how SIG Release
+    works overall—including our expectations regarding technical skills,
+    communications/responsiveness, and reliability
+- working on SIG Release items that improve our interactions with Testgrid,
+  cleaning up libraries, etc.
+  - these efforts require interacting and pairing with Release Managers and
+    Associates
+
+There is currently no one accepted process for identifying or selecting
+candidate Release Manager Associates.
+
+## Identification
+
+In the past, we have done the following to consider candidates:
+
+- A candidate expresses interest to a Release Manager or SIG lead
+- A Release Manager or SIG lead recognizes the work of the candidate and
+  suggests they join the team
+- A SIG lead reaches out to current Release Team towards the end of the
+  release to seed candidates
+
+## Selection
+
+Once a candidate has been identified, SIG leads or senior Release Managers will
+do the following:
+
+- discuss the expectations of the role with them
+- confirm interest with the candidate
+- discuss the candidate with other Release Managers and SIG leads
+- vote on confirming the candidate as a Release Manager Associate
+
+If confirmed, members of Release Managers group will work with the candidate
+will work with the new Release Manager Associate on [onboarding][onboarding].
+
+## Promotion
+
+(From ["Becoming a Release Manager"][becoming-a-release-manager])
+
+To become a Release Manager, one must first serve as a Release Manager
+Associate. Associates graduate to Release Manager by actively working on
+releases over several cycles and:
+
+- demonstrating the willingness to lead
+- tag-teaming with Release Managers on patches, to eventually cut a release
+  independently
+  - because releases have a limiting function, we also consider substantial
+    contributions to image promotion and other core Release Engineering tasks
+- questioning how Associates work, suggesting improvements, gathering feedback,
+  and driving change
+- being reliable and responsive
+- leaning into advanced work that requires Release Manager-level access and
+  privileges to complete
```

Comment:
Without being a RMA one could meet most of those requirements. I would love to see more on how SIG Release can uplift RMAs. Is there any requirements for a Release Manager to even interact with a RMA? I do see that @puerco added in a branch manager shadow pre-req, but I do think there's more to SIG Release beyond just branch management.

It's possible I'm looking at RMA the wrong way, but I assumed the role was a a growth role that experienced leads / chairs could help mentor into RM, for example by:

- checking in with RMAs
- assigning issues
- pair-programming
- asking feedback
- sharing their thought process
- offloading that "one quick lgtm/approval" to the RMA and why (it takes 2 minutes vs. 30 seconds)

I by no means expect that level of mentorship all the time, but today, it is non-existent. I also want to be very clear, I don't want to discount the original list, I agree that independent work / drive and the above are all necessary.

Lastly, it's possible, with the TBD RMA period, my feedback goes away. There would be no need to uplift RMAs and there's an explicit prioritization of independent drive / collaborative work. If so, that should be explicitly stated that the RMA journey, that it is really just a trial for RM and there's no difference but permissions.

Reply:
> Without being a RMA one could meet most of those requirements.

This is to some degree true, but for many tasks, you need elevated permissions (e.g. at least to read logs). There are some tasks that anyone can do, like Go updates and various other updates, but those tasks are not too common.

> Is there any requirements for a Release Manager to even interact with a RMA?

Yes. It's documented on the Release Managers page: https://kubernetes.io/releases/release-managers/#release-managers

From the linked page:

> Release Managers are responsible for:
> ...
> * Supporting Release Manager Associates and contributors through actively participating in the Buddy program
>   * Check in monthly with Associates and delegate tasks, empower them to cut releases, and mentor
>   * Being available to support Associates in onboarding new contributors e.g., answering questions and suggesting appropriate work for them to do

There are two major problems with this though:

* Availability of Release Managers in 2021/2022 has been a bit spotty (and that's totally okay, stuff happens!)
* To my knowledge, we never really had a streamlined process for this. For example, who is mentoring who. So far, it mostly works that RM brings some tasks on Slack and waits for someone to eventually pick them up. Now, because of the first point, we were running a little bit low on tasks, so there's that too

> It's possible I'm looking at RMA the wrong way, but I assumed the role was a a growth role that experienced leads / chairs could help mentor into RM, for example by:

I think that's a solid list of tasks and that's mostly how is it supposed to work. Maybe it would be nice to bring this up for discussion at some SIG Release meeting.